src/coreclr/src/tools/Common/TypeSystem/IL/EcmaMethodIL.cs

```diff
@@ -64,8 +64,8 @@ public override byte[] GetILBytes()
             if (_ilBytes != null)
                 return _ilBytes;
 
-            byte[] ilBytes = _methodBody.GetILBytes();
-            return (_ilBytes = ilBytes);
+            Interlocked.CompareExchange(ref _ilBytes, _methodBody.GetILBytes(), null);
+            return _ilBytes;
```

Comment:
Why is it important for this method to use `CompareExchange` but other methods in this type, say `GetLocals` are okay with not using it?

Reply:
The JIT compares IL stream pointers for inlining decisions, so we need synchronization here otherwise we can get different codegen when a race condition hits.

We haven't seen any determinism bugs with the other APIs like GetLocals and GetExceptionRegions, but it would probably be better to use CompareExchange operations too. Looking at the code however, I haven't seen any evidence that these 2 would cause determinism issues.

@SrivastavaAnubhav could you fix the 2 other APIs to use CompareExchange too?

Reply:
Is it ok to just do a CompareExchange at the very end and just deal with the extra construction? @jkotas do you have an opinion on this?

Reply:
Yes, that's fine.